Graphics/GraphicsEngineVulkan/src/PipelineStateCacheVkImpl.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright 2019-2021 Diligent Graphics LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence),
+ *  contract, or otherwise, unless required by applicable law (such as deliberate
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental,
+ *  or consequential damages of any character arising as a result of this License or
+ *  out of the use or inability to use the software (including but not limited to damages
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and
+ *  all other commercial damages or losses), even if such Contributor has been advised
+ *  of the possibility of such damages.
+ */
+
+#include "pch.h"
+#include "PipelineStateCacheVkImpl.hpp"
+#include "RenderDeviceVkImpl.hpp"
+#include "VulkanTypeConversions.hpp"
+#include "DataBlobImpl.hpp"
+
+namespace Diligent
+{
+
+PipelineStateCacheVkImpl::PipelineStateCacheVkImpl(IReferenceCounters*                 pRefCounters,
+                                                   RenderDeviceVkImpl*                 pRenderDeviceVk,
+                                                   const PipelineStateCacheCreateInfo& CreateInfo) :
+    // clang-format off
+    TPipelineStateCacheBase
+    {
+        pRefCounters,
+        pRenderDeviceVk,
+        CreateInfo,
+        false
+    }
+// clang-format on
+{
+    VkPipelineCacheCreateInfo VkPipelineStateCacheCI{};
+    VkPipelineStateCacheCI.sType = VK_STRUCTURE_TYPE_PIPELINE_CACHE_CREATE_INFO;
+
+    if (CreateInfo.pCacheData != nullptr && CreateInfo.CacheDataSize > 0)
+    {
+        VkPipelineStateCacheCI.initialDataSize = CreateInfo.CacheDataSize;
+        VkPipelineStateCacheCI.pInitialData    = CreateInfo.pCacheData;
```

Comment:
In theory, if the data isn't valid, then the cache data will be ignored.
But in practice, some (older) drivers neglect to verify the UUID correctly.
```
// Provided by VK_VERSION_1_0
typedef struct VkPipelineCacheHeaderVersionOne {
    uint32_t                        headerSize;
    VkPipelineCacheHeaderVersion    headerVersion;
    uint32_t                        vendorID;
    uint32_t                        deviceID;
    uint8_t                         pipelineCacheUUID[VK_UUID_SIZE];
} VkPipelineCacheHeaderVersionOne;
```

Reply:
Yes, I agree. We should check the driver/vendor version and reject the cache if there is a mismatch